include/arch/arm/arch/64/mode/kernel/vspace.h

```diff
@@ -130,15 +114,24 @@ static inline exception_t performASIDPoolInvocation(asid_t asid, asid_pool_t *po
 
 static inline exception_t performASIDPoolInvocation(asid_t asid, asid_pool_t *poolPtr, cte_t *cte)
 {
+    asid_map_t asid_map;
     cap_page_global_directory_cap_ptr_set_capPGDMappedASID(&cte->cap, asid);
     cap_page_global_directory_cap_ptr_set_capPGDIsMapped(&cte->cap, 1);
-    poolPtr->array[asid & MASK(asidLowBits)] =
-        PGDE_PTR(cap_page_global_directory_cap_get_capPGDBasePtr(cte->cap));
-
+    asid_map = asid_map_asid_map_vspace_new(
```

Comment:
nitpicking: go for c99?
```suggestion
    asid_map_t asid_map = asid_map_asid_map_vspace_new(
```